lib/awful/layout/init.lua

```diff
@@ -164,8 +164,10 @@ function layout.parameters(t, screen)
         gap_single_client = t.gap_single_client
     end
 
-    local min_clients       = gap_single_client and 1 or 2
-    local useless_gap       = t and (#clients >= min_clients and t.gap or 0) or 0
+    local useless_gap = 0
+    if (t and (gap_single_client or (#clients >= 2 and layout.get(screen).name ~= "max"))) then
```

Comment:
It would make sense to also match the `fullscreen` layout, I found this **really** ugly way (it seems lua doesn't support general alternation):
```(#clients >= 2 and not string.find(awful.layout.get(client.screen).name, "[mf][au][xl]"))```